### PR TITLE
Add support for F5 TMOS devices in pyntc

### PR DESCRIPTION
**Pull request by mzbroch**
Support for F5 TMOS under pyntc library,

### CODE REVIEW

File: pyntc/devices/f5_device.py [hunk 1]
```diff
@@ -0,0 +1,480 @@
...
+        uptime = uptime % 60
+        seconds = uptime
+
+        return '%02d:%02d:%02d:%02d' % (days, hours, mins, seconds)
+
+    @property
+    def facts(self):
+        facts = {
+            'uptime': self._get_uptime(),
+            'vendor': 'F5 Networks',
```
Comment (jmcgill298) [inline pyntc/devices/f5_device.py]: This should match self.vendor per other device classes

File: pyntc/devices/f5_device.py [hunk 2]
```diff
@@ -0,0 +1,480 @@
...
+        result = self.api_handler.tm.sys.software.volumes.volume.exists(
+            name=volume_name)
+
+        return result
+
+    def _get_active_volume(self):
+        """Gets name of active volume on the device
+
+        Returns:
+            str - name of active volume
```
[the remark this answers is not in the document]
Comment (mzbroch) [reply pyntc/devices/f5_device.py]: There is always one current volume (the one that system is actively running on), so volumes list can not be an empty list, so this should always return a string (in ideal world :-) )

File: pyntc/devices/f5_device.py [hunk 3]
```diff
@@ -0,0 +1,480 @@
...
+            md5sum_result = md5sum_output.commandResult
+            md5sum_result = md5sum_result.split()[0]
+
+        return md5sum_result
+
+    @staticmethod
+    def _file_copy_local_file_exists(filepath):
+        return os.path.isfile(filepath)
+
+    def _file_copy_local_md5(self, filepath, blocksize=2 ** 20):
```
[the remark this answers is not in the document]
Comment (mzbroch) [reply pyntc/devices/f5_device.py]: Correct.

File: pyntc/devices/f5_device.py [hunk 4]
```diff
@@ -0,0 +1,480 @@
...
+
+        while time.time() < end_time:
+            time.sleep(5)
+            try:
+                self._reconnect()
+                volume = self.api_handler.tm.sys.software.volumes.volume.load(
+                    name=volume_name)
+                if hasattr(volume, 'active') and volume.active is True:
+                    return True
+            except Exception:
```
[the remark this answers is not in the document]
Comment (mzbroch) [reply pyntc/devices/f5_device.py]: In this case we try to catch as many as possible to go through reboot moment when the response from the device might be unstable and unpredictable.

File: pyntc/devices/f5_device.py [hunk 5]
```diff
@@ -0,0 +1,480 @@
...
+
+    def file_copy(self, src, dest=None, **kwargs):
+        if dest:
+            raise NotImplementedError(
+                "Support only for images - destination is always /shared/images")
+
+        self._upload_image(image_filepath=src)
+
+    def file_copy_remote_exists(self, src, dest=None, **kwargs):
+        if dest:
```
[the remark this answers is not in the document]
Comment (mzbroch) [reply pyntc/devices/f5_device.py]: Currently this method is limited only to upload images into "/shared/images" directory - I put additional if statement to distinguish and be more transparent about it.

File: pyntc/devices/f5_device.py (hunk 1 above)
Comment (mzbroch) [reply pyntc/devices/f5_device.py]: Changed this, however vendor assignment is still static as I don't know the method to get it dynamically (is there a real need for it?)

**Pull request merged**